src/clp_ffi_js/ir/UnstructuredIrStreamReader.cpp

```diff
@@ -147,13 +147,12 @@ auto UnstructuredIrStreamReader::deserialize_stream() -> size_t {
             }
         }
 
-        auto log_viewer_event{LogEventWithLevel<four_byte_encoded_variable_t>(
-                log_event.get_timestamp(),
-                log_event.get_utc_offset(),
-                message,
-                log_level
+        auto log_event_with_filter_data{LogEventWithFilterData<UnstructuredLogEvent>(
+                log_event,
+                log_level,
+                log_event.get_timestamp()
         )};
-        m_encoded_log_events.emplace_back(std::move(log_viewer_event));
+        m_encoded_log_events.emplace_back(std::move(log_event_with_filter_data));
```

Comment:
As the rabbit suggested, we could emplace directly?

Reply:
If I'm not getting Kirk's comment wrong, I think he meant we could get rid of the separated variable declaration and directly write:
```
        m_encoded_log_events.emplace_back(
                log_event,
                log_level,
                log_event.get_timestamp()
        );
```

Reply:
Yeah, that's what I mean.